addons/website_mass_mailing/static/src/snippets/s_newsletter_block/options.js

```diff
@@ -1,28 +1,55 @@
-/** @odoo-module **/
-
+import { xml } from "@odoo/owl";
 import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
 import { _t } from "@web/core/l10n/translation";
+import { registry } from "@web/core/registry";
 import { renderToElement } from "@web/core/utils/render";
-import options from "@web_editor/js/editor/snippets.options.legacy";
+import { SelectTemplate, SnippetOption } from "@web_editor/js/editor/snippets.options";
+
+
+export class NewsletterBlock extends SelectTemplate {
+    constructor() {
+        super(...arguments);
+        this.containerSelector = '> .container, > .container-fluid, > .o_container_small';
+        this.selectTemplateWidgetName = 'newsletter_template_opt';
+    }
+}
 
-options.registry.mailing_list_subscribe = options.Class.extend({
-    init() {
-        this._super(...arguments);
-        this.orm = this.bindService("orm");
-    },
+export class NewsletterMailingList extends SnippetOption {
+    constructor() {
+        super(...arguments);
+        this.dialog = this.env.services.dialog;
+        this.orm = this.env.services.orm;
+        this.website = this.env.services.website;
+    }
+
+    /**
+     * @override
+     */
+    async willStart() {
+        await super.willStart();
+        this.renderContext.recaptcha_public_key = await this.orm.call(
+            "ir.config_parameter",
+            "get_param",
+            ["recaptcha_public_key"],
+        );
+    }
 
     /**
      * @override
      */
-    onBuilt() {
-        this._super(...arguments);
-        if (this.mailingLists.length) {
-            this.$target.attr("data-list-id", this.mailingLists[0][0]);
+    async onBuilt() {
+        await super.onBuilt(...arguments);
+        if (!this.renderContext.mailingLists) {
+            await this._loadMailingLists();
+        }
+
+        if (this.renderContext.mailingLists.length) {
+            this.$target.attr("data-list-id", this.renderContext.mailingLists[0][0].toString());
         } else {
-            this.call("dialog", "add", ConfirmationDialog, {
+            this.dialog.add(ConfirmationDialog, {
                 body: _t("No mailing list found, do you want to create a new one? This will save all your changes, are you sure you want to proceed?"),
                 confirm: () => {
-                    this.trigger_up("request_save", {
+                    this.website.websiteRootInstance.trigger_up("request_save", {
```

Comment:
Have you tested this? I don't think this actually works as you're sending that event inside the iframe, which does not handle `request_save`. I'd be really surprised if it does work. However, you should be able to call this.env.requestSave to achieve the same thing.